cargo build fails on Windows 11

Hi, 

when I try to add the cargo package on my project and try to build rav1d, I get this error:

```Caused by:
  process didn't exit successfully: `C:\Users\alexpacio\git\netter\rdisplay\target\debug\build\rav1d-49eda93025ada57b\build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-changed=src\x86\.\cdef_sse.asm
  cargo:rerun-if-changed=src\x86\.\itx_sse.asm
  cargo:rerun-if-changed=src\x86\.\msac.asm
  cargo:rerun-if-changed=src\x86\.\pal.asm
  cargo:rerun-if-changed=src\x86\.\refmvs.asm
  cargo:rerun-if-changed=src\x86\.\cdef_avx2.asm
  cargo:rerun-if-changed=src\x86\.\itx_avx2.asm
  cargo:rerun-if-changed=src\x86\.\itx_avx512.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration_avx2.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain_sse.asm
  cargo:rerun-if-changed=src\x86\.\ipred_sse.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter_sse.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration_sse.asm
  cargo:rerun-if-changed=src\x86\.\mc_sse.asm
  cargo:rerun-if-changed=src\x86\.\cdef_avx512.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain_avx2.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain_avx512.asm
  cargo:rerun-if-changed=src\x86\.\ipred_avx2.asm
  cargo:rerun-if-changed=src\x86\.\ipred_avx512.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter_avx2.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter_avx512.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration_avx512.asm
  cargo:rerun-if-changed=src\x86\.\mc_avx2.asm
  cargo:rerun-if-changed=src\x86\.\mc_avx512.asm
  cargo:rerun-if-changed=src\x86\.\cdef16_sse.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain16_sse.asm
  cargo:rerun-if-changed=src\x86\.\ipred16_sse.asm
  cargo:rerun-if-changed=src\x86\.\itx16_sse.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter16_sse.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration16_sse.asm
  cargo:rerun-if-changed=src\x86\.\mc16_sse.asm
  cargo:rerun-if-changed=src\x86\.\ipred16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\cdef16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\cdef16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\filmgrain16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\ipred16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\itx16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\itx16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\loopfilter16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\looprestoration16_avx512.asm
  cargo:rerun-if-changed=src\x86\.\mc16_avx2.asm
  cargo:rerun-if-changed=src\x86\.\mc16_avx512.asm
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration_avx2.o"   
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\msac.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\msac.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\pal.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\pal.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\refmvs.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\refmvs.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration_sse.o"     
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration16_sse.o" 
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc16_sse.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc16_sse.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\cdef16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\cdef16_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\filmgrain16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\filmgrain16_avx512.o"       
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\ipred16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\ipred16_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\itx16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\itx16_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\loopfilter16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\loopfilter16_avx512.o"     
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\looprestoration16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\looprestoration16_avx512.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc16_avx2.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc16_avx2.o"
  running: "nasm" "-fwin64" "-g" "-g" "-fwin64" "-IC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out/" "-Isrc/" "C:\\Users\\alexpacio\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\rav1d-1.0.0\\src\\x86\\.\\mc16_avx512.asm" "-o" "C:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\mc16_avx512.o"
  OUT_DIR = Some(C:\Users\alexpacio\git\netter\rdisplay\target\debug\build\rav1d-7e3f52b41da46462\out)
  TARGET = Some(x86_64-pc-windows-msvc)
  HOST = Some(x86_64-pc-windows-msvc)
  cargo:rerun-if-env-changed=AR_x86_64-pc-windows-msvc
  AR_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=AR_x86_64_pc_windows_msvc
  AR_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_AR
  HOST_AR = None
  cargo:rerun-if-env-changed=AR
  AR = None
  cargo:rerun-if-env-changed=VCINSTALLDIR
  VCINSTALLDIR = None
  cargo:rerun-if-env-changed=VSTEL_MSBuildProjectFullPath
  VSTEL_MSBuildProjectFullPath = None
  cargo:rerun-if-env-changed=VSCMD_ARG_VCVARS_SPECTRE
  VSCMD_ARG_VCVARS_SPECTRE = None
  cargo:rerun-if-env-changed=WindowsSdkDir
  WindowsSdkDir = None
  cargo:rerun-if-env-changed=WindowsSDKVersion
  WindowsSDKVersion = None
  cargo:rerun-if-env-changed=LIB
  LIB = None
  PATH = Some(C:\Users\alexpacio\git\netter\rdisplay\target\debug\deps;C:\Users\alexpacio\git\netter\rdisplay\target\debug;C:\Users\alexpacio\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\lib;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Program Files\PuTTY\;C:\Program Files\dotnet\;C:\Program Files\Go\bin;C:\Program Files\CMake\bin;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Program Files\NASM\;C:\Users\alexpacio\.cargo\bin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Program Files\PuTTY\;C:\Program Files\dotnet\;C:\Program Files\Go\bin;C:\Program Files\CMake\bin;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Program Files\NASM\;)
  cargo:rerun-if-env-changed=INCLUDE
  INCLUDE = None
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=ARFLAGS_x86_64-pc-windows-msvc
  ARFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=ARFLAGS_x86_64_pc_windows_msvc
  ARFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_ARFLAGS
  HOST_ARFLAGS = None
  cargo:rerun-if-env-changed=ARFLAGS
  ARFLAGS = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rustc-link-lib=static=rav1dasm
  cargo:rustc-link-search=native=C:\Users\alexpacio\git\netter\rdisplay\target\debug\build\rav1d-7e3f52b41da46462\out
  cargo:rustc-link-lib=static=rav1dasm
  cargo:rustc-link-lib=static=oldnames
  cargo:rustc-link-search=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\lib\x64
  cargo:rustc-link-search=C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64
  cargo:rustc-link-search=C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\um\x64
  OUT_DIR = Some(C:\Users\alexpacio\git\netter\rdisplay\target\debug\build\rav1d-7e3f52b41da46462\out)
  TARGET = Some(x86_64-pc-windows-msvc)
  OPT_LEVEL = Some(0)
  HOST = Some(x86_64-pc-windows-msvc)
  cargo:rerun-if-env-changed=VCINSTALLDIR
  VCINSTALLDIR = None
  cargo:rerun-if-env-changed=VSTEL_MSBuildProjectFullPath
  VSTEL_MSBuildProjectFullPath = None
  cargo:rerun-if-env-changed=VSCMD_ARG_VCVARS_SPECTRE
  VSCMD_ARG_VCVARS_SPECTRE = None
  cargo:rerun-if-env-changed=WindowsSdkDir
  WindowsSdkDir = None
  cargo:rerun-if-env-changed=WindowsSDKVersion
  WindowsSDKVersion = None
  cargo:rerun-if-env-changed=LIB
  LIB = None
  PATH = Some(C:\Users\alexpacio\git\netter\rdisplay\target\debug\deps;C:\Users\alexpacio\git\netter\rdisplay\target\debug;C:\Users\alexpacio\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\lib;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Program Files\PuTTY\;C:\Program Files\dotnet\;C:\Program Files\Go\bin;C:\Program Files\CMake\bin;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Program Files\NASM\;C:\Users\alexpacio\.cargo\bin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Program Files\PuTTY\;C:\Program Files\dotnet\;C:\Program Files\Go\bin;C:\Program Files\CMake\bin;C:\Program Files\LLVM\bin;C:\Program Files (x86)\LLVM\bin;C:\Program Files\NASM\;)
  cargo:rerun-if-env-changed=INCLUDE
  INCLUDE = None
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some(cmpxchg16b,fxsr,sse,sse2,sse3)
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  getopt.c
  c1: fatal error C1083: Cannot open source file: 'tools/compat/getopt.c': No such file or directory

  --- stderr


  error occurred: Command "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.39.33519\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-I" "include/compat" "-W4" "-FoC:\\Users\\alexpacio\\git\\netter\\rdisplay\\target\\debug\\build\\rav1d-7e3f52b41da46462\\out\\b7820ac75fe99709-getopt.o" "-c" "tools/compat/getopt.c" with args cl.exe did not execute successfully (status code exit code: 2).
```
  
  
 I am running on Windows 11 23H2 and using the latest rust/cargo 1.81.0 version.
  
 Can you help me to figure out why?

--- a/build.rs
+++ b/build.rs
@@ -292,10 +292,10 @@ mod asm {
             nasm.files(asm_file_paths);
             #[cfg(debug_assertions)]
             nasm.flag("-g");
-            #[cfg(all(debug_assertions, not(windows)))]
-            nasm.flag("-Fdwarf");
-            #[cfg(all(debug_assertions, windows))]
-            nasm.flag("-fwin64");
+            nasm.flag(match os {
+                "windows" => "-fwin64",
+                _ => "-Fdwarf",
+            });
             nasm.flag(&format!("-I{}/", out_dir.to_str().unwrap()));
             nasm.flag("-Isrc/");
             let obj = nasm.compile_objects().unwrap_or_else(|e| {

--- a/build.rs
+++ b/build.rs
@@ -331,41 +331,4 @@ fn main() {
     {
         asm::main();
     }
-
-    // NOTE: we rely on libraries that are only distributed for Windows so
-    // targeting Windows/MSVC is not supported when cross compiling.
-    #[cfg(all(target_os = "windows", target_env = "msvc"))]
-    {
-        use cc::windows_registry;
-        use std::env;
-
-        let os = env::var("CARGO_CFG_TARGET_OS").expect("missing CARGO_CFG_TARGET_OS");
-        let target = env::var("TARGET").expect("missing TARGET");
-        if os == "windows" {
-            // for sprintf, snprintf, etc.
-            println!("cargo:rustc-link-lib=static=oldnames");
-            let tool = windows_registry::find_tool(&target, "cl.exe")
-                .expect("couldn't find cl.exe; are the Visual Studio C++ tools installed?");
-            let lib_paths = &tool
-                .env()
-                .iter()
-                .find(|(key, _val)| key == "LIB")
-                .expect("LIB path not found")
-                .1;
-            for path in lib_paths.to_str().unwrap().split(';') {
-                if path != "" {
-                    println!("cargo:rustc-link-search={path}");
-                }
-            }
-
-            let getopt = "getopt";
-            cc::Build::new()
-                .files([&"tools/compat/getopt.c"])
-                .include("include/compat")
-                .debug(cfg!(debug_assertions))
-                .compile(&getopt);
-            // cc automatically outputs the following line
-            // println!("cargo:rustc-link-lib=static={getopt}");
-        }
-    }
 }

--- a/build.rs
+++ b/build.rs
@@ -292,6 +292,7 @@ mod asm {
             nasm.files(asm_file_paths);
             #[cfg(debug_assertions)]
             nasm.flag("-g");
+            #[cfg(debug_assertions)]
             nasm.flag(match os {
                 "windows" => "-fwin64",
                 _ => "-Fdwarf",

--- a/tools/Cargo.toml
+++ b/tools/Cargo.toml
@@ -22,6 +22,9 @@ cfg-if = "1.0.0"
 libc = "0.2"
 rav1d = { path = "../", version = "1.0.0", default-features = false }
 
+[build-dependencies]
+cc = "1.0.79"
+
 [features]
 default = ["asm", "asm_arm64_dotprod", "asm_arm64_i8mm", "bitdepth_8", "bitdepth_16"]
 asm = ["rav1d/asm"]

--- a/tools/build.rs
+++ b/tools/build.rs
@@ -1,0 +1,38 @@
+fn main() {
+    // NOTE: we rely on libraries that are only distributed for Windows so
+    // targeting Windows/MSVC is not supported when cross compiling.
+    #[cfg(all(target_os = "windows", target_env = "msvc"))]
+    {
+        use cc::windows_registry;
+        use std::env;
+
+        let os = env::var("CARGO_CFG_TARGET_OS").expect("missing CARGO_CFG_TARGET_OS");
+        let target = env::var("TARGET").expect("missing TARGET");
+        if os == "windows" {
+            // for sprintf, snprintf, etc.
+            println!("cargo:rustc-link-lib=static=oldnames");
+            let tool = windows_registry::find_tool(&target, "cl.exe")
+                .expect("couldn't find cl.exe; are the Visual Studio C++ tools installed?");
+            let lib_paths = &tool
+                .env()
+                .iter()
+                .find(|(key, _val)| key == "LIB")
+                .expect("LIB path not found")
+                .1;
+            for path in lib_paths.to_str().unwrap().split(';') {
+                if path != "" {
+                    println!("cargo:rustc-link-search={path}");
+                }
+            }
+
+            let getopt = "getopt";
+            cc::Build::new()
+                .files([&"../tools/compat/getopt.c"])
+                .include("../include/compat")
+                .debug(cfg!(debug_assertions))
+                .compile(&getopt);
+            // cc automatically outputs the following line
+            // println!("cargo:rustc-link-lib=static={getopt}");
+        }
+    }
+}

--- a/tools/build.rs
+++ b/tools/build.rs
@@ -1,38 +1,52 @@
+use std::env;
+
 fn main() {
+    let os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+    let env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+
+    let os = os.as_str();
+    let env = env.as_str();
+
+    match (os, env) {
+        ("windows", "msvc") => {
+            if !cfg!(all(target_os = "windows", target_env = "msvc")) {
+                panic!("Cannot cross compile to *-windows-msvc");
+            }
+        }
+        #[allow(clippy::needless_return)]
+        _ => return,
+    }
+
     // NOTE: we rely on libraries that are only distributed for Windows so
     // targeting Windows/MSVC is not supported when cross compiling.
     #[cfg(all(target_os = "windows", target_env = "msvc"))]
     {
         use cc::windows_registry;
-        use std::env;
 
-        let os = env::var("CARGO_CFG_TARGET_OS").expect("missing CARGO_CFG_TARGET_OS");
-        let target = env::var("TARGET").expect("missing TARGET");
-        if os == "windows" {
-            // for sprintf, snprintf, etc.
-            println!("cargo:rustc-link-lib=static=oldnames");
-            let tool = windows_registry::find_tool(&target, "cl.exe")
-                .expect("couldn't find cl.exe; are the Visual Studio C++ tools installed?");
-            let lib_paths = &tool
-                .env()
-                .iter()
-                .find(|(key, _val)| key == "LIB")
-                .expect("LIB path not found")
-                .1;
-            for path in lib_paths.to_str().unwrap().split(';') {
-                if path != "" {
-                    println!("cargo:rustc-link-search={path}");
-                }
+        // for sprintf, snprintf, etc.
+        let target = env::var("TARGET").unwrap();
+        println!("cargo:rustc-link-lib=static=oldnames");
+        let tool = windows_registry::find_tool(&target, "cl.exe")
+            .expect("couldn't find cl.exe; are the Visual Studio C++ tools installed?");
+        let lib_paths = &tool
+            .env()
+            .iter()
+            .find(|(key, _val)| key == "LIB")
+            .expect("LIB path not found")
+            .1;
+        for path in lib_paths.to_str().unwrap().split(';') {
+            if path != "" {
+                println!("cargo:rustc-link-search={path}");
             }
-
-            let getopt = "getopt";
-            cc::Build::new()
-                .files([&"../tools/compat/getopt.c"])
-                .include("../include/compat")
-                .debug(cfg!(debug_assertions))
-                .compile(&getopt);
-            // cc automatically outputs the following line
-            // println!("cargo:rustc-link-lib=static={getopt}");
         }
+
+        let getopt = "getopt";
+        cc::Build::new()
+            .files([&"../tools/compat/getopt.c"])
+            .include("../include/compat")
+            .debug(cfg!(debug_assertions))
+            .compile(&getopt);
+        // cc automatically outputs the following line
+        // println!("cargo:rustc-link-lib=static={getopt}");
     }
 }

